File: responser-backend/src/main/java/com/responser/backend/model/Response.java
package com.responser.backend.model;

import jakarta.persistence.*;
import lombok.*;

/**
 * Response
 *
 * @author SIE
 */
@Getter
@Setter
@EqualsAndHashCode(callSuper = true)
@RequiredArgsConstructor
@Entity
@Table(name = "responses")
public class Response extends AbstractEntity{

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "resource_id")
    private String resourceId;

    @Column(name = "response_id")
    private String responseId;

    private Byte rating;

    private String text;
}

File: responser-chrome-extension/src/app/model/Response.ts
import {User} from "./User";
import moment, {Moment} from "moment";

export class Response {

    id: string;

    user: User;

    resourceId: string;

    responseId: string;

    rating: number;

    text: string;

    creationDate: Moment;

    updateDate: Moment;

    public static deserialize(data: any): Response {
        return Object.assign(new Response(), data, {
            user: User.deserialize(data.user),
            createdTimestamp: moment(data.createdTimestamp),
            updateDate: moment(data.updateDate)
        });
    }
}